src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -984,6 +984,14 @@ trait Contexts { self: Analyzer =>
         }
       )
     }
+    // If ambiguous import resolves to the conflicting symbol, allow it.
+    // Typically, implicit import scala._ conflicts with definitions in scala packages.
+    // For that case, check if the root import dealiases to the definition.
+    private def resolveAmbiguity(name: Name, defSym: Symbol, impSym: Symbol, imp: ImportInfo): Boolean = {
+      defSym == impSym || imp.isRootImport && {
+        impSym.isAliasType && impSym.info.typeSymbol.fullName == defSym.fullName
```

Comment:
@retronym Help. I didn't have time to improve this yet.

Reply:
I'm having trouble paging in the context for this from my mental swap space... could you elaborate a little?

Reply:
If root `import scala._` alias `Seq` collides with `Seq` defined in enclosing package, recognize that they are the same. There is related code for when two imports collide.